instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -33,98 +36,204 @@ type gRPCContextKey struct{}
 type gRPCContext struct {
 	messagesReceived int64
 	messagesSent     int64
+	requestSize      int64
+	responseSize     int64
+	metricAttrs      []attribute.KeyValue
 }
 
 // NewServerHandler creates a stats.Handler for gRPC server.
 func NewServerHandler(opts ...Option) stats.Handler {
 	h := &serverHandler{
 		config: newConfig(opts),
+		r:      &handler{},
 	}
 
-	h.tracer = h.config.TracerProvider.Tracer(
+	h.r.tracer = h.config.TracerProvider.Tracer(
 		instrumentationName,
 		trace.WithInstrumentationVersion(SemVersion()),
 	)
+	h.r.meter = h.config.MeterProvider.Meter(
+		instrumentationName,
+		metric.WithInstrumentationVersion(Version()),
+		metric.WithSchemaURL(semconv.SchemaURL),
+	)
+	var err error
```

Comment:
Should we return the error here so users can handle it?

Reply:
IIRC the interceptor had the same behavior. IMO it feels like an error should be returned here and the logger should be used for steady state failures.